### PR TITLE
Create Category Model

### PR DESCRIPTION
**Pull request by michaelwehrley**
Create Category Model

* Category attributes exist.
* Add in Rails Admin.
* Update rspec gem and .rspec.
* Add specs.

[Finishes: #91531724]

### CODE REVIEW

**Comment by ericcf** [comment]
Looks like this can't be merged until you rebase.

File: app/models/category.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Are sets of chapters that display text, image, and an active
+# state to the user so they know what module they are in.
+class Category < ActiveRecord::Base
+  validates :active, inclusion: { in: [true, false] }
```
Comment (eschlange) [inline app/models/category.rb]: @ericcf Mike and I are wondering if the "inclusion" parameter is necessary.  Would leaving that out be exactly the same as what exists now? Is it good practice for code documentation to use this notation for boolean typed attributes?

Comment (ericcf) [inline app/models/category.rb]: I can't find what used to be there any longer. However, what Mike has is the Rails convention for validating a boolean. Without this, if you tried to set it to `nil` for instance, you'd get a Postgres exception when trying to set the column to `NULL`.

**Comment by eschlange** [comment]
Wouldn't adding ```null: false``` be a little more readable?

**Comment by ericcf** [comment]
I'm not sure I understand. Adding that where?

**Comment by eschlange** [comment]
Nevermind~ I found the specification you mention here: http://edgeguides.rubyonrails.org/active_record_validations.html. I was thinking of the migration specifications and not model validation.

**Comment by michaelwehrley** [comment]
For clarification, if you want to validate the presence of a boolean field you should use one of the following validations:
```
validates :boolean_field_name, presence: true
validates :boolean_field_name, inclusion: { in: [true, false] }
validates :boolean_field_name, exclusion: { in: [nil] }
```
http://edgeguides.rubyonrails.org/active_record_validations.html

**Comment by ericcf** [comment]
Thanks. The `presence` validation failed in previous versions of Rails: http://guides.rubyonrails.org/v4.1.8/active_record_validations.html#presence

Nice to know it works as expected now!

**Pull request merged**